hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableAccountStore.java

```diff
@@ -213,4 +214,39 @@ public Set<AccountID> modifiedAccountsInState() {
     public Set<Bytes> modifiedAliasesInState() {
         return aliases().modifiedKeys();
     }
+
+    /**
+     * Gets the original value associated with the given accountId before any modifications were made to
+     * it. The returned value will be {@code null} if the accountId does not exist.
+     *
+     * @param id The accountId. Cannot be null, otherwise an exception is thrown.
+     * @return The original value, or null if there is no such accountId in the state
+     * @throws NullPointerException if the accountId is null.
+     */
+    @Nullable
+    public Account getOriginalValue(@NonNull final AccountID id) {
+        requireNonNull(id);
+        // Get the account number based on the account identifier. It may be null.
+        final var accountOneOf = id.account();
+        final Long accountNum =
+                switch (accountOneOf.kind()) {
```

Comment:
IIRC I've seen switch statements for account vs. alias in a couple of places now. Can we not call the existing logic instead of duplicating it here, or is this different enough that it requires new code?

Reply:
yes, it should call `getOriginalValue()` everywhere